tests/components/slack/test_notify.py

```diff
@@ -1,77 +1,116 @@
 """Test slack notifications."""
+import copy
 import logging
+from typing import List
 from unittest.mock import AsyncMock, Mock, patch
 
 from _pytest.logging import LogCaptureFixture
 import aiohttp
 from slack.errors import SlackApiError
 
+from homeassistant.components import notify
+from homeassistant.components.slack import DOMAIN
 from homeassistant.components.slack.notify import (
     CONF_DEFAULT_CHANNEL,
     SlackNotificationService,
-    async_get_service,
 )
-from homeassistant.const import CONF_API_KEY, CONF_ICON, CONF_USERNAME
+from homeassistant.const import CONF_API_KEY, CONF_ICON, CONF_PLATFORM, CONF_USERNAME
 from homeassistant.helpers.typing import HomeAssistantType
+from homeassistant.setup import async_setup_component
 
 MODULE_PATH = "homeassistant.components.slack.notify"
 
-
-async def test_get_service(hass: HomeAssistantType, caplog: LogCaptureFixture):
-    """Test async_get_service with exceptions."""
-    config = {
-        CONF_API_KEY: "12345",
-        CONF_DEFAULT_CHANNEL: "channel",
-    }
-
-    with patch(MODULE_PATH + ".aiohttp_client") as mock_session, patch(
-        MODULE_PATH + ".WebClient"
+DEFAULT_CONFIG = {
+    notify.DOMAIN: [
+        {
+            CONF_PLATFORM: DOMAIN,
+            CONF_API_KEY: "12345",
+            CONF_DEFAULT_CHANNEL: "channel",
+        }
+    ]
+}
+
+
+def filter_log_records(caplog: LogCaptureFixture) -> List[logging.LogRecord]:
+    """Filter all unrelated log records."""
+    return [
+        rec for rec in caplog.records if rec.name.endswith(f"{DOMAIN}.{notify.DOMAIN}")
+    ]
+
+
+async def test_setup(hass: HomeAssistantType, caplog: LogCaptureFixture):
+    """Test setup slack notify."""
+    config = DEFAULT_CONFIG
+
+    with patch(
+        MODULE_PATH + ".aiohttp_client",
+        **{"async_get_clientsession.return_value": (session := Mock())},
+    ), patch(
+        MODULE_PATH + ".WebClient",
+        return_value=(client := AsyncMock()),
     ) as mock_client, patch(
-        MODULE_PATH + ".SlackNotificationService"
+        MODULE_PATH + ".SlackNotificationService",
```

Comment:
We shouldn't patch the platform code. We should patch the library code and assert calls to the library client mock.

Reply:
Ok. I've pushed a new commit.